pkg/config/datafileprojectconfig/config.go

```diff
@@ -33,6 +33,8 @@ var datafileVersions = map[string]struct{}{
 // DatafileProjectConfig is a project config backed by a datafile
 type DatafileProjectConfig struct {
 	datafile             string
+	hostForODP           string
+	publicKeyForODP      string
```

Comment:
I see that a logic to get all segments used in datafile audiences is missing. We need it later when we call `fetchSegments`. See https://github.com/optimizely/swift-sdk/blob/ec83c87d53b17ea58f8c73eb581b27fa52b6305c/Sources/Data%20Model/ProjectConfig.swift#L151

Reply:
I was originally holding it off for the next PR but have now implemented it in this one. Please have a look at the new changes.